specs/_features/eip7732/beacon-chain.md

```diff
@@ -0,0 +1,796 @@
+
+At any given slot, the status of the blockchain's head may be either 
+- A block from a previous slot (e.g. the current slot's proposer did not submit its block). 
+- An *empty* block from the current slot (e.g. the proposer submitted a timely block, but the builder did not reveal the payload on time). 
+- A full block for the current slot (both the proposer and the builder revealed on time). 
+
+## Constants
+
+### Payload status
+
+| Name | Value | 
+| - | - | 
+| `PAYLOAD_ABSENT` | `uint8(0)` |
+| `PAYLOAD_PRESENT` | `uint8(1)` | 
+| `PAYLOAD_WITHHELD` | `uint8(2)` | 
+| `PAYLOAD_INVALID_STATUS` | `uint8(3)` |
+
+## Preset
+
+### Misc
+
+| Name | Value | 
+| - | - | 
+| `PTC_SIZE` | `uint64(2**9)` (=512)  # (New in EIP-7732) |
+
+### Domain types
+
+| Name | Value |
+| - | - |
+| `DOMAIN_BEACON_BUILDER`     | `DomainType('0x1B000000')`  # (New in EIP-7732)|
+| `DOMAIN_PTC_ATTESTER`       | `DomainType('0x0C000000')`  # (New in EIP-7732)|
+
+### Max operations per block
+
+| Name | Value |
+| - | - |
+| `MAX_PAYLOAD_ATTESTATIONS` | `2**2` (= 4)  # (New in EIP-7732) |
+
+## Containers
+
+### New containers
+
+#### `PayloadAttestationData`
+
+```python
+class PayloadAttestationData(Container):
+    beacon_block_root: Root
+    slot: Slot
+    payload_status: uint8
+```
+
+#### `PayloadAttestation`
+
+```python
+class PayloadAttestation(Container):
+    aggregation_bits: Bitvector[PTC_SIZE]
+    data: PayloadAttestationData
+    signature: BLSSignature
+```
+
+#### `PayloadAttestationMessage`
+
+```python
+class PayloadAttestationMessage(Container):
+    validator_index: ValidatorIndex
+    data: PayloadAttestationData
+    signature: BLSSignature
+```
+
+#### `IndexedPayloadAttestation`
+
+```python
+class IndexedPayloadAttestation(Container):
+    attesting_indices: List[ValidatorIndex, PTC_SIZE]
+    data: PayloadAttestationData
+    signature: BLSSignature
+```
+
+#### `SignedExecutionPayloadHeader`
+
+```python
+class SignedExecutionPayloadHeader(Container):
+    message: ExecutionPayloadHeader
+    signature: BLSSignature
+```
+    
+#### `ExecutionPayloadEnvelope`
+
+```python
+class ExecutionPayloadEnvelope(Container):
+    payload: ExecutionPayload
+    builder_index: ValidatorIndex
+    beacon_block_root: Root
+    blob_kzg_commitments: List[KZGCommitment, MAX_BLOB_COMMITMENTS_PER_BLOCK]
+    payload_withheld: boolean
+    state_root: Root
+```
```

Comment:
Side note that this one would benefit from EIP-7495 SSZ StableContainer as `payload_withheld` could be dropped and `payload` become `Optional[ExecutionPayload]`

Reply:
Indeed, I'm waiting for that to be adopted in Electra and I'll rebase on top of it